postgresqlPackages.anonymizer: init at 1.3.1; add me & osnyx to flyingcircus team

Ma27 opened the pull request:
## Description of changes

* Adds me and @osnyx to the flyingcircus team (we discussed this directly, but please approve here as well for the record :)). Also needs an approval from one of the other team members (cc @dpausp @ctheune @frlan @leona-ya )
* Adds the postgresql extension [pg_anonymizer](https://gitlab.com/dalibo/postgresql_anonymizer).


## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Commit:
maintainers: add osnyx & ma27 to team flyingcircus

Commit:
postgresqlPackages.anonymizer: init at 1.3.1

osnyx reviewed (approved):
I approve the part where I am added as a maintainer; I have not (yet) looked at the postgres anonymizer part though.

osnyx reviewed (approved):
Now also approving the full packaging and testing code.

Ma27 commented:
Updated to 1.3.1, the latest version of the package being available.

leona-ya reviewed (changes requested):
The compilation of the `postgresqlJitPackages.anonymizer` package fails for me (for all psql versions) with:

```
install -d /nix/store/cdsqi903x9lik27rxh9yxsm5zczwvxyw-postgresql_anonymizer-1.3.1/bin
install -m 0755 bin/pg_dump_anon.sh /nix/store/cdsqi903x9lik27rxh9yxsm5zczwvxyw-postgresql_anonymizer-1.3.1/bin
/nix/store/mb488rr560vq1xnl10hinnyfflcrd51n-coreutils-9.4/bin/install -c -m 644 .//anon.control '/nix/store/cdsqi903x9lik27rxh9yxsm5zczwvxyw-postgresql_anonymizer-1.3.1/share/postgresql/extension/'
/nix/store/mb488rr560vq1xnl10hinnyfflcrd51n-coreutils-9.4/bin/install -c -m 644 .//anon/*  '/nix/store/cdsqi903x9lik27rxh9yxsm5zczwvxyw-postgresql_anonymizer-1.3.1/share/postgresql/extension/anon/'
/nix/store/mb488rr560vq1xnl10hinnyfflcrd51n-coreutils-9.4/bin/install -c -m 755  anon.so '/nix/store/cdsqi903x9lik27rxh9yxsm5zczwvxyw-postgresql_anonymizer-1.3.1/lib/'
/nix/store/mb488rr560vq1xnl10hinnyfflcrd51n-coreutils-9.4/bin/mkdir -p '/nix/store/cdsqi903x9lik27rxh9yxsm5zczwvxyw-postgresql_anonymizer-1.3.1/lib/bitcode/anon'
/nix/store/mb488rr560vq1xnl10hinnyfflcrd51n-coreutils-9.4/bin/mkdir -p '/nix/store/cdsqi903x9lik27rxh9yxsm5zczwvxyw-postgresql_anonymizer-1.3.1/lib/bitcode'/anon/
/nix/store/mb488rr560vq1xnl10hinnyfflcrd51n-coreutils-9.4/bin/install -c -m 644 anon.bc '/nix/store/cdsqi903x9lik27rxh9yxsm5zczwvxyw-postgresql_anonymizer-1.3.1/lib/bitcode'/anon/./
cd '/nix/store/cdsqi903x9lik27rxh9yxsm5zczwvxyw-postgresql_anonymizer-1.3.1/lib/bitcode' && llvm-lto -thinlto -thinlto-action=thinlink -o anon.index.bc anon/anon.bc
/nix/store/4vzal97iq3dmrgycj8r0gflrh51p8w1s-bash-5.2p26/bin/bash: line 1: llvm-lto: command not found
make: *** [/nix/store/8hkjv5idbsal1rf5b8hmc43rx2k0gzqs-postgresql-13.14/lib/pgxs/src/makefiles/pgxs.mk:235: install] Error 127
```

I don't know whether pganon supports JIT at all, but we should either fix that or mark as broken / remove pganon from the jit packages.

Without JIT it lgtm.

Ma27 commented:
@leona-ya seems like adding `llvm` for that to `nativeBuildInputs` is sufficient, seems to be working fine as well.

For the record, stdenv doesn't need extra configuration, when using JIT, there's already an LLVM stdenv passed into the extensions.

Commit:
pg-dump-anon: init at 1.3.1

This is a Go program inside the sources of `postgresql_anonymizer` that
allows to perform database dumps, but with anonymized data. I figured
that it's a little awkward to have a client program to be part of the
extension package.

So I decided to create a second package called `pg-dump-anon`. Since
it's one repository, both share `version` & `src`.

Also extended the VM test to make sure we're getting properly anonymized
data when dumping with `pg_dump_anon`.

Ma27 commented:
Pushed another commit which adds the `pg_dump_anon` program. This can be used on a remote database, i.e. without the extension part, hence the new package. I decided against using multiple outputs here since it's a Go package and the Go builder doesn't seem composable, so I'd have to split up the build processes anyways.

Also extended the test to make sure the program does what it advertises, providing database dumps, but with data being anonymized.

cc @leona-ya @osnyx

osnyx commented on nixos/tests/pg_anonymizer.nix:
```diff
@@ -0,0 +1,94 @@
+            2,Salazar,
+            \.
+
+        in the given dump (the commas are tabs in case of pg_dump).
```
It's a bit wild that pg_dump and pg_dump_anon appear to use different formats, but I guess that's not your fault.

marsam commented on pkgs/servers/sql/postgresql/ext/anonymizer.nix:
```diff
@@ -0,0 +1,32 @@
+  '';
+
+  meta = pg-dump-anon.meta // {
+    description = "postgresql_anonymizer is an extension to mask or replace personally identifiable information (PII) or commercially sensitive data from a PostgreSQL database.";
```
nitpick
```suggestion
    description = "Extension to mask or replace personally identifiable information (PII) or commercially sensitive data from a PostgreSQL database";
```

marsam reviewed (approved):
LGTM

Ma27 committed:
postgresqlPackages.anonymizer: update description

Co-authored-by: Mario Rodas <marsam@users.noreply.github.com>
